docs/generate_command_docs.py

```diff
@@ -0,0 +1,148 @@
+"""This module generate reference docs for pebble CLI commands."""
+
+import logging
+import os
+import re
+import subprocess
+import typing
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+AUTOMATED_START_MARKER = "<!-- START AUTOMATED OUTPUT -->"
+AUTOMATED_STOP_MARKER = "<!-- END AUTOMATED OUTPUT -->"
+
+TEMPLATE = """\
+(reference_pebble_{command}_command)=
+# {command} command
+
+{description}
+
+## Usage
+
+<!-- START AUTOMATED OUTPUT -->
+```{{terminal}}
+   :input: command
+```
+<!-- END AUTOMATED OUTPUT -->
+"""
+
+
+def get_all_commands() -> typing.List[typing.Tuple[str, str]]:
+    process = subprocess.run(
+        ["go", "run", "../cmd/pebble", "help", "--all"],
+        text=True,
+        capture_output=True,
+        check=True,
+    )
+    pattern = r"^\s{4}(\w+(?:-\w+)?)(?:\s{2,}|$)([\w\s\'\-\.\n]+?)(?=\n\s{4}|$)"
```

Comment:
To me this regex seems almost unreadable / unmaintainable. It seems like we're processing such a simple thing here: find the first word and the rest of the line for all lines that start with 4 spaces. Can we do this:

```python
return sorted(line.split(maxsplit=1) for line in process.stdout.splitlines()
              if line.startswith('    '))
```

Avoiding a fancy regex here is simpler and much more readable.

Also, for sorting lists of lists or lists of tuples on the first element, you don't need `key=lambda x: x[0]`, because when the items being sorted are lists or tuples, Python automatically compares the first element first, then the second, and so on.